doc/modules/neighbors.rst

```diff
@@ -510,3 +510,211 @@ the model from 0.81 to 0.82.
 
   * :ref:`sphx_glr_auto_examples_neighbors_plot_nearest_centroid.py`: an example of
     classification using nearest centroid with different shrink thresholds.
+
+
+.. _nca:
+
+Neighborhood Components Analysis
+================================
+
+.. sectionauthor:: William de Vazelhes <william.de-vazelhes@inria.fr>
+
+Neighborhood Components Analysis (NCA, :class:`NeighborhoodComponentsAnalysis`)
+is a distance metric learning algorithm which aims to improve the accuracy of
+nearest neighbors classification compared to the standard Euclidean distance.
+The algorithm  directly  maximizes  a stochastic  variant  of  the
+leave-one-out k-nearest neighbors (KNN) score on the training set.  It can also
+learn a low-dimensional linear  embedding  of  data  that  can  be used for
+data  visualization and fast classification.
+
+.. |nca_illustration_1| image:: ../auto_examples/neighbors/images/sphx_glr_plot_nca_illustration_001.png
+   :target: ../auto_examples/neighbors/plot_nca_illustration.html
+   :scale: 50
+
+.. |nca_illustration_2| image:: ../auto_examples/neighbors/images/sphx_glr_plot_nca_illustration_002.png
+   :target: ../auto_examples/neighbors/plot_nca_illustration.html
+   :scale: 50
+
+.. centered:: |nca_illustration_1| |nca_illustration_2|
+
+In the above illustrating figure, we consider some points from a randomly
+generated dataset. We focus on the stochastic KNN classification of point no. 3,
+the thickness of a bond representing a softmax distance hence the weight of the
```

Comment:
maybe clarify a bit this sentence because at this point there is not so much context on the method for the user to rely on

proposition:
"The thickness of a link between sample 3 and another point is proportional to their distance, and can be seen as the relative weight (or probability) that a stochastic nearest neighbor prediction rule would assign to this point."

it is possible to refer to the mathematical formulation section for details, but maybe it is better not to do this to avoid confusing users

Reply:
Yes I agree, I'll go for your proposition it's clearer
I'll add a link to the mathematical formulation, they could leave it alone and read the examples before, depending on what they would prefer